config/resolve.go

```diff
@@ -1,23 +1,26 @@
 package config
 
 import (
+	"context"
 	"crypto/tls"
 	"crypto/x509"
 	"fmt"
+	"io/ioutil"
 	"net/http"
 	"os"
 
+	"github.com/awslabs/smithy-go/logging"
+
 	"github.com/aws/aws-sdk-go-v2/aws"
 	awshttp "github.com/aws/aws-sdk-go-v2/aws/transport/http"
-	"github.com/awslabs/smithy-go/logging"
 )
 
 // resolveDefaultAWSConfig will write default configuration values into the cfg
 // value. It will write the default values, overwriting any previous value.
 //
 // This should be used as the first resolver in the slice of resolvers when
 // resolving external configuration.
-func resolveDefaultAWSConfig(cfg *aws.Config, cfgs configs) error {
+func resolveDefaultAWSConfig(ctx context.Context, cfg *aws.Config, cfgs configs) error {
```

Comment:
This resolve probably should be hoisted out of the resolvers and called directly by `LoadDefaultConfig` before the caller's passed in functional options are processed.

Reply:
resolveDefaultAWSConfig was not hoisted as, Loaders need to be processed before this resolve occurs.

If this resolve function is hoisted the loader resolve functions would not resolve credentials correctly from env config, shared config coz anonymous credentials are already set by `resolveDefaultAWSConfig` function.